src/app/records/PresentationNodeLeaf.tsx

```diff
@@ -13,17 +15,103 @@ export default function PresentationNodeLeaf({
   ownedItemHashes,
   completedRecordsHidden,
   redactedRecordsRevealed,
+  sortRecordProgression,
 }: {
   node: DimPresentationNodeLeaf;
   ownedItemHashes?: Set<number>;
   completedRecordsHidden: boolean;
   redactedRecordsRevealed: boolean;
+  sortRecordProgression: boolean;
 }) {
+  const newNode = { ...node };
+  if (sortRecordProgression) {
+    /**
+     * RECORDS
+     */
+    if (node.records) {
+      newNode.records = orderBy(
+        newNode.records,
+        (record) => {
+          /**
+           * Triumph is already completed so move it to back of list.
+           * State is a bitmask where the ones place is for when the
+           * record has been redeemed. The other checks move completed
+           * but not redeemed to the back.
+           */
+          if (
+            record.recordComponent.state & DestinyRecordState.RecordRedeemed ||
+            record.recordComponent.state & DestinyRecordState.CanEquipTitle ||
+            !record.recordComponent.state
+          ) {
+            return -1;
+          }
+
+          // check which key is used to track progress
+          let objectives;
+          if (record.recordComponent.intervalObjectives) {
+            objectives = record.recordComponent.intervalObjectives;
+          } else if (record.recordComponent.objectives) {
+            objectives = record.recordComponent.objectives;
+          } else {
+            // its a legacy triumph so it has no objectives
+            return 0;
+          }
+
+          /**
+           * Iterate through the objectives array and return the average
+           * of its % progress.
```

Comment:
It's not correct to average interval objectives - they form "segments" of an overall progress bar. Interval objectives should be summed.

Reply:
My bad, my comment is wrong. I don't do this, I take the sum. I've changed the comment.

Reply:
"interval objectives" are like one big progress bar, split up into segments. So that needs to have the progress and the totals summed up to find the overall progress.

Some other objectives show things like "the fastest time you've completed this task" and thus don't really make sense in terms of progress.

Reply:
So I need to write specific code for the time trial triumphs in the `DimMetric`s then? They are the only triumphs I can find on my account that track time. Even then they still have a `progress` and `completionValue` and the metric still turns yellow when marked completed. I have not found a triumph that does not `progress` and `completionValue`.

Reply:
Sounds good then. Let's not keep discussing on a closed PR.